### PR TITLE
Remove usage of TaskCache

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Localization/RequestCultureProvider.cs [hunk 1]
```diff
@@ -13,6 +13,12 @@ namespace Microsoft.AspNetCore.Localization
     public abstract class RequestCultureProvider : IRequestCultureProvider
     {
         /// <summary>
+        /// Result that indicates that this instance of <see cref="RequestCultureProvider" /> could not determine the
+        /// request culture.
+        /// </summary>
+        protected static readonly Task<ProviderCultureResult> NullResult = Task.FromResult(default(ProviderCultureResult));
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Localization/RequestCultureProvider.cs]: This removes the last use of `Task<T>` from our code base (and we can remove the package). Given the pattern requires returning this, it made sense to put the field on the base type. I can alternatively make it an `internal` field or put it in an `.Internal` type.

Comment (Eilon) [inline src/Microsoft.AspNetCore.Localization/RequestCultureProvider.cs]: Hmm this is probably fine. Can't do any harm really. Or maybe it should have a more specific name, though? Is `NullProviderCultureResult` too long? I think it reads better in the code, though.

**Comment by pranavkm** [comment]
> Is NullProviderCultureResult too long? 

Nah, seems fine given it's narrow use. 🆙 📅

**Comment by hishamco** [comment]
I already made similar PR #377, but I confused little bit, because we need to do similar things in all the repos that uses the `TaskCache`

**Comment by pranavkm** [comment]
@hishamco you want to cache the `Task<ProviderCultureResult>` somewhere which was missing in your PR.

**Comment by hishamco** [comment]
Yep, I planned to make it static and store somewhere else, anyhow your PR looks fine 😄

**Pull request merged**